### ISSUE TITLE
Failed compilation due to inexisting output directory makes further compilations impossible (greyed out) until restart

### ISSUE THREAD

**Issue by e-dervieux** [texstudio-org/texstudio#3750]
### Environment
- TeXstudio: 4.8.1 (git 4.8.1)
- Qt: 6.7.1 R
- OS: MacOS Sonoma 14.2.1 (23C71)
- TeX distribution: TeX Live 2023

This bug may be related to:
https://github.com/texstudio-org/texstudio/issues/3371
https://github.com/texstudio-org/texstudio/issues/3254

### Use case

Having a compilation command with an output directory, for instance:
```
lualatex -synctex=1 -output-directory out -interaction=nonstopmode TeXFileName.tex
```
if the directory `out` does not exist, a popup appears saying that the compilation is not possible.

![image](https://github.com/user-attachments/assets/7daa94b9-3180-4f9c-9d04-7c39436eca14)

### Expected behavior

After this popup, we should be able to make further compilations, i.e. pressing F5 or F6, or clicking the "compile" or "compile and preview" buttons.

![image](https://github.com/user-attachments/assets/8cc6c1f1-3799-4980-98dc-8dc059633f3b)

### Actual behavior

After this popup, the "compile" and "compile and preview" buttons are greyed out, and the F5 and F6 shortcuts become ineffective. TeXStudio becomes completely stuck, and only quitting it and relaunching it makes the compilations options available again.

![image](https://github.com/user-attachments/assets/2b7390c0-ee13-43e6-8e01-d2e73c815f32)

### How to reproduce

Create a simple TeX file and try to compile it with the above-given compilation command.

**Comment by octaeder** [comment]
can't reproduce. Screenshots from message pane

![image](https://github.com/user-attachments/assets/e5454339-6e3e-41df-86b3-8534b2bd6719)

and explorer

![image](https://github.com/user-attachments/assets/604c5c48-7b73-414b-872e-35b6ae974896)

TeXstudio 4.8.2 (git 4.8.2alpha3-8-g54a467b95)
Using Qt Version 6.7.2, compiled with Qt 6.7.2 R
Win10

So I expect it to work in 4.8.1.

**Comment by e-dervieux** [comment]
Tested in 4.8.1 even with only a blank file -> BUG as reported.

I tried to grab the alpha3 [here](https://github.com/texstudio-org/texstudio/releases) but neither the M1 ZIP version nor the DMG one did start: the app just crashes silently so I cannot test it.

**Comment by sunderme** [comment]
my guess is that the search path for pdf and log is not set.
Both need to be set to "out" as well.
<img width="678" alt="grafik" src="https://github.com/user-attachments/assets/bea42cd9-da26-46ff-9809-ee6deb9a060a">

**Comment by octaeder** [comment]
I didn't set additional search paths. While my first test may be a little bit special in that it is an msys2/linux shell, I now made a second test with 4.8.0 which I have still installed on the same win10 system. Again, I found no issue. Folder out is created where the tex file is located.

**Comment by e-dervieux** [comment]
Just to be clear: the issue is arising when the folder "out" does **NOT** exist, and you try to compile anyway. If the folder "out" do exist where the `.tex` file is located, there is no issue.

@octaeder, you wrote "Folder out is created where the tex file is located", you mean that if the "out" folder does not exist in your system, compiling the `.tex` file creates it? In this case, I can assure you that TS does not behave this way in MacOS.

**Note:** @sunderme, I do have "out" in my additional Log and PDF search paths.

**Comment by sunderme** [comment]
the folder out needs to be generated by lualatex, txs has no input in this.
Please debug the issue on command line on your system. Maybe OSX does not allow the folder generation ?

**Comment by e-dervieux** [comment]
OK, so I ran `lualatex -synctex=1 -output-directory out -interaction=nonstopmode test.tex` from the terminal, wherein `test.tex` is an empty file. This yields the following output:
```
This is LuaHBTeX, Version 1.16.0 (TeX Live 2023) 
 restricted system commands enabled.
! I can't write on file `test.log'.
Please type another transcript file name
! Emergency stop
!  ==> Fatal error occurred, no output PDF file produced!%
```
But even though it comes from some MacOSesque oddity, there is still a bug in TS which should not be stuck until next restart, I think.

**Note:** if it may help from a debugging point of view, in TS, once the compilation is stuck for the above-mentioned reasons (error message + greyed out buttons), if I set another `.tex` file as explicit master file, the compilation options enable again. I can then unset this latter file's explicit master role and try again to compile my `test.tex` file (which unsurprisingly leads to the same error message and greying out of the compilation buttons).
